### ISSUE TITLE
Support `fixture_ref()` in tuple of parameters

### ISSUE THREAD

**Issue by smarie** [smarie/python-pytest-cases#47]

**Comment by smarie** [comment]
It seems that if the fixture represents the whole tuple, it works, but it is not possible to use the reference for only one element of the tuple. To investigate, in order to see if this is really desirable.

**Comment by smarie** [comment]
To add a more explicit description: this does not work yet because the fixture reference is only used as a **part** of the parameters tuple:

```python
import pytest
from pytest_cases import pytest_parametrize_plus, fixture_ref

@pytest.fixture
def test():
    return ['a', 'b', 'c']

@pytest_parametrize_plus('p,q', [('c', 1),
                                 (fixture_ref(test), 1)
                                 ])
def test_prints(p, q):
    assert p in ['a', 'b', 'c']
    assert q == 1
```

**Comment by yashtodi94** [comment]
Please add this feature. Would be of great use in testing different combinations in a complex function.

Also, the tests would be more reliable as one will then be able to do:

`assert p == 'a'` instead of `assert p in ['a', 'b', 'c']`

**Comment by smarie** [comment]
Sorry my example code was wrong. I updated it to reflect what should actually be the behaviour (it uses pytest-harvest to get the synthesis)

**Comment by yashtodi94** [comment]
Yes that is exactly what I am referring to. This will make the test code's syntax (and functionality) similar to how `@pytest.mark.parametrize` actually works. Will help in testing out multiple combinations of parameters. Also, saves a person from writing multiple tests just to test the combinations.

What do you think?

**Comment by smarie** [comment]
perfect. I'll see what I can do.

**Comment by smarie** [comment]
Done in 1.11.0. But I warn you: the names become hard to follow ! :)

Let me know if it works for you.
